# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../busybox/legacy_sus.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

usleep() was deprecated by SUSv3 and newer LIBCs are starting to enforce that
deprecation.

--- ./include/libbb.h.orig	2010-09-16 17:02:15.000000000 -0400
+++ ./include/libbb.h	2010-09-16 17:07:27.000000000 -0400
@@ -1751,6 +1751,12 @@
 #define isgraph_asciionly(a) ((unsigned)((a) - 0x21) <= 0x7e - 0x21)
 #define isprint_asciionly(a) ((unsigned)((a) - 0x20) <= 0x7e - 0x20)
 
+#define usleep(usec)	bb_usleep(usec)
+static ALWAYS_INLINE int bb_usleep(useconds_t usec)
+{
+	struct timespec t = { usec/1000000, (usec%1000000)*1000L };
+	return nanosleep(&t, NULL);
+}
 
 POP_SAVED_FUNCTION_VISIBILITY
 
